src/Utils.ts

```diff
@@ -0,0 +1,39 @@
+
+/**
+ * Compares two components.
+ * @remarks
+ *
+ * this recursive function is responsible for comparing two component.
+ *
+ * @param x - the first component that we are going to compare.
+ * @param y - the second component that we are going to compare.
+ * @returns true, if both components are identical; false, if the components are not identical.
+ *
+ */
+const backwardsCheck = (x: any, y: any): boolean => {
+  for (const p in y) {
+    if (Object.prototype.hasOwnProperty.call(y, p) && !Object.prototype.hasOwnProperty.call(x, p)) {return false;}
+  }
+  return true;
+};
+const compareComponents = (x: any, y: any): boolean => {
+  if (!(x instanceof Object) || !(y instanceof Object)) {return false;}
+  // if they are not strictly equal, they both need to be Objects
+  for (const p in x) {
+    if (!Object.prototype.hasOwnProperty.call(x, p)) {continue;}
+
+    if (!Object.prototype.hasOwnProperty.call(y, p)) {return false;}
+    // allows to compare x[ p ] and y[ p ] when set to undefined
+
+    if (x[String(p)] === y[String(p)] || p.startsWith('x-')) {continue;}
```

Comment:
Please describe why you extension is skipped.

Reply:
because extensions have different values for objects that are equal (duplicated.) If I don't skip the extensions this function always returns false.

Reply:
I meant to write a comment about 😅  Sorry